juntagrico_billing/entity/payment.py

```diff
@@ -17,6 +17,7 @@ class Payment(JuntagricoBaseModel):
     paid_date = models.DateField(_('Payment date'), null=True, blank=True)
     amount = models.FloatField(_('Amount'), null=False, blank=False, default=0.0)
     private_notes = models.TextField(_('Notes not visible to {}').format(Config.vocabulary('member_pl')), null=True, blank=True)
+    unique_id = models.CharField(_('Unique Id'), max_length=50, null=True, blank=True)
```

Comment:
warum wurde kein unique constraint definiert im meta teil?

Reply:
Die Unique Id verwenden wir nur, für aus Files importierte Zahlungen um zu verhindern, dass ein Zahlungsfile doppelt eingelesen wird. Ist nicht mandatory und kann darum auch nicht vom Schema her unique sein.

Reply:
null solte von einem unique constrain ignoriert werden von dem her wäre das schon noch gut

Reply:
Aber ein leeres CharField wird als empty string gespeichert und dann würde das unique wieder greifen, siehe https://stackoverflow.com/questions/454436/unique-fields-that-allow-nulls-in-django.

Habe das Gefühl, das führt zu einigem Aufwand und der Nutzen des unique settings ist gering.

Reply:
Verstehe deinen Vorschlag nicht ganz. Du meinst, das Feld im admin nichht anzeigen?
Und warum das blank==True rausnehmen, das braucht es doch bei Char-Feldern wenn die leer sein können?
Ich kann das Feld auch einfach umbenennen, damit es nicht mir unique_id heisst. Z.B. import_id, wäre das eine Lösung?

Reply:
ein commit sagt mehr als tausend worte.
so sollte es klappen. das problem war nicht, dass etwas unique im namen hatte und dass keinen unique constraint hatte, sondern dass du den defacto unique constraint via datenbank abfrage gelöst hast, was zu unnötigen datnebank abfragen führt und die unique id war acuh in der datenadministration exposed wo irgend ein unbeholfener user sie einfach anpassen oder vergeben kann und das ganze system schrottet oder zumindes inkonsistent macht. so ist nun sichergestellt, dass egal wo im code ein payment mit unique id erfasst wird die auch wirklich unique ist